### ISSUE TITLE
Re-compute Bounds in Slicer

### ISSUE THREAD

**Issue by Garyfallidis** [fury-gl/fury#150]
When the slicer goes up or down we need to be recomputing the bounds because some of the slices are not immediately visible.

**Comment by Saransh0905** [comment]
Hey! I find fury pretty interesting and I want to contribute to it.So I want to start by solving this issue.
Can you guide me how to start?
Thanks

**Comment by skoudoro** [comment]
Hi @Saransh0905,

Your first step is to reproduce the error. Can you reproduce it? 

- If No,  I recommend you to start looking at the Tutorial and Examples on FURY website, look at [the contributing page](http://fury.gl/latest/contributing_link.html) and how VTK works.

- If Yes, look how to compute bounds in VTK and update the slicer function.

Hope it helps

**Comment by Saransh0905** [comment]
Thanks @skoudoro . Is there any IRC channel where I can communicate and ask for help.

**Comment by skoudoro** [comment]
> Is there any IRC channel where I can communicate and ask for help.

on the community page of the website, there is a slack link.

**Comment by Saransh0905** [comment]
@Garyfallidis  I  tried to reproduce the issue by adding a loop to  move the slicer in Z direction but I didn't see anything wrong.
![Screenshot](https://user-images.githubusercontent.com/48832748/74866934-b8839200-5379-11ea-85e7-e77a5192980b.png)

What should I do to reproduce the error?

**Comment by Garyfallidis** [comment]
Hi @Saransh0905 to see this problem you have to connect the slicer to a slider. As we do here
https://github.com/dipy/dipy/blob/master/dipy/viz/panel.py

**Comment by Garyfallidis** [comment]
@Saransh0905 you will have to call ComputeBounds in this line to resolve the problem
https://github.com/fury-gl/fury/blob/13809a10cb4a6753f166ef547ac6a087357189ad/fury/actor.py#L167

**Comment by Garyfallidis** [comment]
The problem basically is that when you start moving the slicer up and down some times it goes out of the field of view and the slices are not visible.

**Comment by Garyfallidis** [comment]
@skoudoro can you provide @Saransh0905 with the one-liner to compute bounds and update the field of view. @Saransh0905 let me know if something is not clear.

**Comment by Saransh0905** [comment]
Thank you so much @Garyfallidis I understood the problem

**Comment by Saransh0905** [comment]
Now to solve this we have to recalculate the bounds .
Is this GetBounds different from ComputeBounds you mentioned https://github.com/fury-gl/fury/blob/13809a10cb4a6753f166ef547ac6a087357189ad/fury/actor.py#L2033

**Comment by ChenCheng0630** [comment]
Hi, I see there is no further updates on this issue, so plan to contribute to solving this issue. 

I think this is the problem @Garyfallidis trying to solve right?
<img width="392" alt="Screenshot 2020-03-05 at 10 54 33 PM" src="https://user-images.githubusercontent.com/44283369/75999063-ad656000-5f3c-11ea-97e3-36a6d5c6e63a.png">

**Comment by skoudoro** [comment]
Hi @ChenCheng0630,

It seems you can reproduce the issue which is good! So feel free to fix this issue! Nobody seems to work on it right now

**Comment by ChenCheng0630** [comment]
Hi @skoudoro ,
After investigating on this issue for one whole day, I finally got a better understanding of the problem. 

I tried to print out the bounds of the slice while moving the slider, the bounds are all correct, they stay as what they should be no matter how I move the slider. 

Once I stopped moving the slider, and started to interact with the scene(rotate and zoom), the entire slice became visible immediately.  Hence I went to investigate VTK source code and user manual, and realised the method `renderer.ResetCameraClippingRange()` is called every time when we interact with the scene. 

It turns out I think this is a really a clipping range problem not a bounds problem. Parts of the slicer gone invisible because the clipping range stay the same while we moving the slider. And if the slicer moves out of the clipping range, it would be invisible. 

```python
def change_slice_x(slider):
    x = int(np.round(slider.value))
    slice_actor.display_extent(x, x, 0, shape[1]-1, 0, shape[2]-1)
    sm.scene.reset_clipping_range()
```
If I call reset camera clipping range right after changing the display extent, then everything would be normal. 

Therefore, I wonder what would a better way to solve the issue? As slicers(actors) have no reference to scene, would not be able to change the clipping range by slicers themselves.  Or is there a way for slicers to manipulate their bounds such that somewhere the method `scene.ResetCameraClippingRange()` would be triggered?
